### PR TITLE
[BUGFIX] ensure handleResponse always called in rest adapter

### PR DESCRIPTION
**Pull request by Gaurav0**
For #6851

### CODE REVIEW

File: packages/-ember-data/tests/integration/adapter/handle-response-test.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+    class TestAdapter extends JSONAPIAdapter {
+      handleResponse(status, headers, payload, requestData) {
+        handleResponseCalled++;
+
+        if (this.isSuccess(status, headers, payload)) {
+          return payload;
+        }
+
+        let errors = this.normalizeErrorResponse(status, headers, payload);
```
Comment (runspired) [inline packages/-ember-data/tests/integration/adapter/handle-response-test.js]: should we just super once we increment?

File: packages/-ember-data/tests/unit/utils/determine-body-promise-test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  });
+
+  test('determineBodyResponse returns the body even if it is not json', function(assert) {
+    assert.expect(1);
+
+    const response = new Response('this is not json', { status: 200 });
+    const bodyPromise = determineBodyPromise(response, {});
+
+    return bodyPromise.then(body => {
+      assert.deepEqual(body, 'this is not json');
```
Comment (runspired) [inline packages/-ember-data/tests/unit/utils/determine-body-promise-test.js]: let's add a description here

File: packages/-ember-data/tests/unit/utils/determine-body-promise-test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+module('Unit | determineBodyPromise', function() {
+  test('determineBodyResponse returns the body when it is present', function(assert) {
+    assert.expect(1);
+
+    const response = new Response('{"data": "foo"}', { status: 200 });
+    const bodyPromise = determineBodyPromise(response, {});
+
+    return bodyPromise.then(body => {
+      assert.deepEqual(body, { data: 'foo' });
```
Comment (runspired) [inline packages/-ember-data/tests/unit/utils/determine-body-promise-test.js]: let's add a description

File: packages/-ember-data/tests/integration/adapter/rest-adapter-test.js [hunk 4]
```diff
@@ -2576,10 +2576,6 @@ module('integration/adapter/rest_adapter - REST Adapter', function(hooks) {
         hash.error(jqXHR, jqXHR.responseText, errorThrown);
       };
 
-      adapter.handleResponse = function(status, headers, payload) {
```
Comment (runspired) [inline packages/-ember-data/tests/integration/adapter/rest-adapter-test.js]: why did we need to delete this? Are we now calling this method *more* often than we used to? (note how this is in a `hasJQuery` branch)

Comment (Gaurav0) [reply packages/-ember-data/tests/integration/adapter/rest-adapter-test.js]: Yes. We are now ensuring we are calling it always, whether fetch or jquery. Let me know if this is not correct.

File: packages/-ember-data/tests/integration/adapter/handle-response-test.js (hunk 1 above)
Comment (Gaurav0) [reply packages/-ember-data/tests/integration/adapter/handle-response-test.js]: done.

File: packages/-ember-data/tests/integration/adapter/rest-adapter-test.js (hunk 4 above)
Comment (Gaurav0) [reply packages/-ember-data/tests/integration/adapter/rest-adapter-test.js]: Update: I now no longer call it in this situation (a non JSON string; status 200)

**Comment by igorT** [comment]
Do we want this on the LTS?

**Pull request merged**